### ISSUE TITLE
bdist_pex called from tox fails with python 3.12

### ISSUE THREAD

**Issue by anisse** [pex-tool/pex#2413]
When pex is called via bdist_pex from tox, it will fail, but only on python 3.12 (python 3.9, 3.10 and 3.11 work):
```
Failed to spawn a job for /home/runner/work/pex-minimal-repros/pex-minimal-repros/.tox/bundle/bin/python: received exit code 1 during execution of `['/tmp/tmptki_lyu7/pip/bin/python3.12', '-s', '-E', '-m', 'pip', 'install', '-U', 'pip']` while trying to execute `['/tmp/tmptki_lyu7/pip/bin/python3.12', '-s', '-E', '-m', 'pip', 'install', '-U', 'pip']`
```

I made a [repo](https://github.com/anisse/pex-minimal-repros/commit/340a21235865de1b4ef7bfc73b28390e971330c3) to reproduce the issue, and the full output of a failed build can be found here: https://github.com/anisse/pex-minimal-repros/actions/runs/9209146256/job/25333005165#step:4:168

**Comment by jsirois** [comment]
@anisse I won't be back at a keyboard to look at either this or #2412 until May 29th, but I will note that instead of using `bdist_pex`, you could just say `pex --disable-cache . -r requirements.in --resolver-version pip-2020-resolver --pip-version 24.0 -c entry -o entry.pex`. Of course you'd need to do this once per-entrypoint, which is certainly less convenient than `--bdist-all` for a `setup.py` with many console scripts where you want 1 PEX binary per-each. For that though, you might consider using [`conscript`](https://pypi.org/project/conscript/).

**Comment by anisse** [comment]
Thank you for the quick answer, I might try the workaround. There is no emergency and I'll keep the repos up for a while.

**Comment by jsirois** [comment]
@anisse as I indicated in #2413, I won't have time to dig into a fix for this Python 3.12 + `--disable-cache` + `bdist_pex`-specific issue until May 29th, but I did have some extra time today to debug a bit deeper anyhow.

**Comment by jsirois** [comment]
Ok, by experiment, this is fixed by not relying on `ensurepip` and instead modifying `Virtualenv.create` to take a new defaulted argument of `include_pip=False` here and only specifying `--no-pip` / `--without-pip` when it is `False`:
https://github.com/pex-tool/pex/blob/52ce99f183db28c23321026340f17281087fee6d/pex/venv/virtualenv.py#L163-L256

If I remove the `venv.install_pip` here and instead pass `include_pip=True` in the `Virtualenv.create` call, everything works fine:
https://github.com/pex-tool/pex/blob/52ce99f183db28c23321026340f17281087fee6d/pex/pip/installation.py#L102-L103

**Comment by jsirois** [comment]
@anisse this should be solved with Pex 2.8.0:
+ https://github.com/pex-tool/pex/releases/tag/v2.8.0
+ https://pypi.org/project/pex/2.8.0/
